iothub/device/src/Messaging/OutgoingMessage.cs

```diff
@@ -0,0 +1,241 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// The data structure that represents the message that will be sent to IotHub.
+    /// </summary>
+    public class OutgoingMessage
```

Comment:
I'd call it `TelemetryMessage` because I think that is mainly how customers think about it, but I'm open to other ideas.

Reply:
I don't mind TelemetryMessage; however, we'd need to rename other associated classes  as well, right? Eg. TelemetryMessageAcknowledgement?

Reply:
Probably, yes.